ytfzf

```diff
@@ -1134,13 +1142,76 @@ get_history () {
 }
 
 clear_history () {
-	if [ -e "$history_file" ]; then
-		: > "$history_file"
-		printf "History has been cleared\n"
+    case "$1" in
+        search) h_file="${search_history_file}" ;;
+        watch) h_file="${history_file}" ;;
+        *) h_file="${search_history_file}|${history_file}" ;;
+    esac
+
+    exit_code=0
+    IFS="|"
+    for file in $h_file; do
+        if [ "$file" = "$history_file" ]; then
+            name="History"
+        else
+            name="Search History"
+        fi
+        if [ -e "$file" ]; then
+            : > "$file"
+            printf "%s has been cleared\n" "$name"
+        else
+            printf "\033[31m%s file not found, history not cleared\033[0m\n" "$name" >&2
+            exit_code=1
+        fi
+    done
+    exit $exit_code
+}
+
+get_search_history () {
+	if [ "$enable_search_hist" -eq 1 ]; then
+		[ -e "$search_history_file" ] || touch "$search_history_file"
+		#gets history data in reverse order (makes it most recent to least recent)
+		hist_data=$( sed '1!G; h; $!d' "$search_history_file" )
+		[ -z "$hist_data" ] && printf "Search history is empty!\n" >&2 && return 1;
```

Comment:
instead of `return 1` this should be `exit 1`.

Reply:
Maybe we should leave it as it is and let the fzf menu open up empty, since it serves a secondary purpose of being able to be used as another form of query input?

Reply:
well, leaving it as is, is inconsistent with -H when history is empty.

I think it's better to exit, because otherwise you're left with an empty menu which feels buggy, and unintuitive, at least in my opinion. If it said maybe, "Make a search query" it might be ok, but it's still inconsistent with -H, and it would be harder to implement.

Although it is nice to be able to search again from this point, running ytfzf like this would be the same: `ytfzf -q || ytfzf [search]`

Reply:
Hmm, history and search history, although similar in many aspect, do have fundamentally different uses.
With history we are only using fzf as a filter for a list of videos, with search history we are using it as a filter for query's as well as an input for new query's to get a new list of videos.
What I trying to say is that there two different aspects of the user pipeline.
Although maybe a good compromise could be a `allow_empty_search_history` option in config?

Reply:
that would be a good compromise, I see your point of them serving different purposes

my suggestion of how to add it:

```suggestion
if -z "$hist_data" ]; then
    printf "Search history is empty!\n" >&2
    [ $allow_empty_search_history -eq 1 ] && return || exit 1;
fi
```

Reply:
I think it might be a better idea to have both histories return 1 and leave it up to the place where the functions are called to handle how to deal with it.

Reply:
ok